### ISSUE TITLE
TextureRect resets size back to original before _process

### ISSUE THREAD

**Issue by fidli** [godotengine/godot#32058]
**Godot version:** 3.1.1.stable.official

**OS/device including version:** Ubuntu 18.04.2 LTS

**Issue description:** TextureRect resets size before entering process() loop

**Steps to reproduce:** run attached project (including scripts)

**Minimal reproduction project:** see attached script:

basically:
```
extends Node

class_name Minimap
var _dotDiameter = 40

func _init(parent):
	parent.add_child(self)
	var position = Image.new()
	position.load("res://assets/gps_arrow.png")
	
	var postex = ImageTexture.new()
	postex.create_from_image(position)
	get_tree().get_root().get_node("Root/Minimap/Position").texture = postex
	print("init")
	print("s: ", get_tree().get_root().get_node("Root/Minimap/Position").get_size(), " p: ", get_tree().get_root().get_node("Root/Minimap/Position").get("rect_pivot_offset"))
	get_tree().get_root().get_node("Root/Minimap/Position").set_size(Vector2(_dotDiameter, _dotDiameter))
	get_tree().get_root().get_node("Root/Minimap/Position").set("rect_pivot_offset", Vector2(_dotDiameter, _dotDiameter) * 0.5)
	print("s: ", get_tree().get_root().get_node("Root/Minimap/Position").get_size(), " p: ", get_tree().get_root().get_node("Root/Minimap/Position").get("rect_pivot_offset"))


func _process(delta):
	print("process")
	print("s: ", get_tree().get_root().get_node("Root/Minimap/Position").get_size(), " p: ", get_tree().get_root().get_node("Root/Minimap/Position").get("rect_pivot_offset"))

```
gets you this output
```
init
s: (180, 180) p: (0, 0)
s: (40, 40) p: (20, 20)
process
s: (180, 180) p: (20, 20)
...
```
size gets reset after _init() and before _process() loop
[bug.zip](https://github.com/godotengine/godot/files/3590782/bug.zip)

also, if you set it in process, then it stays set

**Comment by KoBeWi** [comment]
It works if you use `set_deferred`, although not sure why is it needed.

**Comment by Megalomaniak** [comment]
I can't even clip it's content or set it's size based on anchoring in editor. Seems to be very buggy, TBH.

**Comment by tom-jk** [comment]
@Megalomaniak can reproduce constant updating in 3.1 thru 3.1.2, not in 3.2 onwards, so that appears fixed.

@fidli try this: start with your MRP (bug.zip project), and replace Minimap.gd with this:

```gdscript
extends Node

class_name Minimap
var mmpos:TextureRect
var _dotDiameter = 40

func _init(parent):
	parent.add_child(self)
	var position = Image.new()
	position.load("res://assets/gps_arrow.png")
	
	var postex = ImageTexture.new()
	postex.create_from_image(position)
	
	mmpos = get_tree().get_root().get_node("Root/Minimap/Position")
	
	mmpos.texture = postex
	print("- init -")
	print("s: ", mmpos.get_size(), " p: ", mmpos.rect_pivot_offset)
	
	# Change the match number to try different methods.
	match 3:
		1:
			# set immediately
			mmpos.set_size(Vector2(_dotDiameter, _dotDiameter))
			mmpos.rect_pivot_offset = Vector2(_dotDiameter, _dotDiameter) * 0.5
		2:
			# set deferred
			mmpos.call_deferred("set_size", Vector2(_dotDiameter, _dotDiameter))
			mmpos.set_deferred("rect_pivot_offset", Vector2(_dotDiameter, _dotDiameter) * 0.5)
		3:
			# wait a frame, then set
			yield(get_tree(), "idle_frame")
			mmpos.set_size(Vector2(_dotDiameter, _dotDiameter))
			mmpos.rect_pivot_offset = Vector2(_dotDiameter, _dotDiameter) * 0.5
		4:
			# wait a frame, then set deferred (pointless? but for completion)
			yield(get_tree(), "idle_frame")
			mmpos.call_deferred("set_size", Vector2(_dotDiameter, _dotDiameter))
			mmpos.set_deferred("rect_pivot_offset", Vector2(_dotDiameter, _dotDiameter) * 0.5)
	
	print("s: ", mmpos.get_size(), " p: ", mmpos.rect_pivot_offset)
	print("- init done - ")


func _process(delta):
	print("s: ", mmpos.get_size(), " p: ", mmpos.get("rect_pivot_offset"))
	pass
```

Using deferred setters (method 2) didn't make a difference, but yielding a frame (method 3) did:

```
- init -
s: (180, 180) p: (0, 0)
s: (40, 40) p: (20, 20)
- init done - 
s: (40, 40) p: (20, 20)
s: (40, 40) p: (20, 20)
etc.
```

(results consistent in 3.1.1, 3.2, 3.2.3.rc3)

**Comment by akien-mga** [comment]
Thanks for the thorough test, closing as fixed in 3.2

**Comment by akien-mga** [comment]
Please open a new issue with a reproduction project.
